crates/stark-felt/src/lib.rs

```diff
@@ -161,11 +173,76 @@ impl Felt {
         Self(self.0.pow(exponent))
```

Comment:
That would leak the implementation to users. They would need to import the `IsUnsignedInteger` from `lambdaworks` to use the `starknet-io` type. At that point, they may as well just use `lambdaworks`.

Reply:
Okay understood.
So we probably need more manual impl, from u8 to u128

Reply:
Do you mean accepting any type that can be converted to `u128`?

Reply:
Yes.

Reply:
It will fail for u8, until the inner trait is changed

Reply:
What I tried was converting to `u128` and that works. It's just a little bit annoying because we need to pass literals with explicit type. There's of course a theoretical penalty on widening the type, but I doubt any slowdown can be measured considering the `pow` itself is expensive.